include/tins/sniffer.h

```diff
@@ -45,6 +45,9 @@
 
 #include <pcap.h>
 
+using pcap_sniffing_method_t = int(*)(pcap_t*, int, pcap_handler, u_char*);
```

Comment:
Fair enough, sounds like a good idea. This way we could actually write tests for `BaseSniffer` at least, which is pretty great. Can you just move the typedef inside `BaseSniffer` and rename it?